src/plugins/discover/public/application/main/services/discover_state.ts

```diff
@@ -375,6 +375,16 @@ export function getDiscoverStateContainer({
    * state containers initializing and subscribing to changes triggering e.g. data fetching
    */
   const initializeAndSync = () => {
+    // This needs to be the first thing that's wired up because initAndSync is pulling the current state from the URL which
+    // might change the time filter and thus needs to re-check whether the saved search has changed.
+    const timefilerUnsubscribe = services.timefilter.getTimeUpdate$().subscribe(() => {
+      savedSearchContainer.update({
```

Comment:
This method contains a lot of logic. What if we apply only `timeRange` and `refreshInterval` changes here instead of the whole `update`?

Reply:
Made an update, is this what you had in mind?

Reply:
Yes, thanks! I think we also need to update `refreshInterval` if it got changed.

Reply:
You're right, could you take another look?